src/app/modules/sample-providers/components/submission-form/submission-form.component.ts

```diff
@@ -1,22 +1,74 @@
-import { Component, OnInit, Output, EventEmitter } from '@angular/core';
+import {
+  Component,
+  OnInit,
+  Output,
+  EventEmitter,
+  Input,
+  ElementRef,
+  ViewChild,
+  HostListener,
+  ChangeDetectionStrategy,
+  ChangeDetectorRef,
+} from '@angular/core';
 import {
   AbstractControl,
   FormBuilder,
   FormGroup,
   Validators,
+  NG_VALUE_ACCESSOR,
+  ControlValueAccessor,
 } from '@angular/forms';
 
 @Component({
   selector: 'app-submission-form',
   templateUrl: './submission-form.component.html',
   styleUrls: ['./submission-form.component.scss'],
+  providers: [
+    {
+      provide: NG_VALUE_ACCESSOR,
+      useExisting: SubmissionFormComponent,
+      multi: true,
+    },
+  ],
+  changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class SubmissionFormComponent implements OnInit {
+export class SubmissionFormComponent implements OnInit, ControlValueAccessor {
   form: FormGroup;
 
+  f: File; // demo file
+
+  @ViewChild('p', { read: ElementRef }) p: ElementRef;
+
   @Output()
   submitForm = new EventEmitter<any>();
 
+  @Input()
+  onChange = (f: File) => {};
+
+  @HostListener('change', ['$event.target.files'])
+  emitFiles(e: FileList) {
+    this.f = e && e.item(0);
+    this.a();
+    this.onChange(this.f);
+  }
+
+  registerOnChange(fn) {
+    this.onChange = fn;
+  }
+
+  registerOnTouched(fn) {}
+
+  /** @ignore */
+  writeValue(value) {
+    if (value === null && this.f) {
+      this.f = null;
+    }
+  }
+  // add audioFile
+  a() {
```

Comment:
Might as well just call this method `addAudioFile` instead of having to leave a comment :) Also, instead of needing to reference the class variable `this.f`, we could simply pass the file value in as a param ie `addAudioFile(file: File)`